src/js/filterRating.js

```diff
@@ -0,0 +1,58 @@
+
+export function handleStarInput(data) {
+
+  const starsFrom = document.querySelectorAll(".starsFrom li");
+  const starsTo = document.querySelectorAll(".starsTo li");
+  
+  let activeIdx = null;
+
+  starsFrom.forEach((starFrom, clickIdx) => {
+    starFrom.addEventListener("click", (e) => {
+      if (activeIdx === clickIdx) {
+        e.target.classList.remove("active");
+        activeIdx = clickIdx - 1;
+      } else {
+        e.target.classList.add("active");
+        activeIdx = clickIdx;
+      }
+  
+      starsFrom.forEach((otherStar, otherIdx) => {
+        if (otherIdx < clickIdx) {
+          otherStar.classList.add("active");
+        } else if (otherIdx > clickIdx) {
+          otherStar.classList.remove("active");
+        }
+      });
+    });
+  });
+
+  starsTo.forEach((starTo, clickIdx) => {
+    starTo.addEventListener("click", (e) => {
+      if (activeIdx === clickIdx) {
+        e.target.classList.remove("active");
+        activeIdx = clickIdx - 1;
+      } else {
+        e.target.classList.add("active");
+        activeIdx = clickIdx;
+      }
+
+      starsTo.forEach((otherStar, otherIdx) => {
+        if (otherIdx < clickIdx) {
+          otherStar.classList.add("active");
+        } else if (otherIdx > clickIdx) {
+          otherStar.classList.remove("active");
+        }
+      });
+    });
+  });
+}
+
+export function filterByRating(data){
+  const valueFrom = document.querySelectorAll('#starFrom .active');
+  const valueTo = document.querySelectorAll('#starTo .active');
```

Comment:
This is fine, but means you're no longer using the `activeIdx` variable as the source of truth, but instead getting the current number of stars from the DOM. This means you're using the DOM as your data store, which is unconventional in modern development because it means that changes to the DOM will also change the underlying logic in many places, not just the place that renders the DOM.

I would have retrieved the `valueFrom` based on the `activeIdx` (`const valueFrom = activeIdx + 1`) and the `valueTo` based on the second variable that should have always been there, yet isn't part of this PR but is added later.